### PR TITLE
Addition of CompyMcNodeFace to Icepack machine files

### PR DESCRIPTION
**Pull request by proteanplanet**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 
Addition of DOE E3SM machine CompyMcNodeFace to standard machines for which Icepack works. 
- [x] Developer(s): 
@proteanplanet on behalf of E3SM
- [x] Suggest PR reviewers from list in the column to the right.
Must be reviewers with access to Compy and who are part of E3SM: @eclare108213 @njeffery 
- [x] Please copy the PR test results link or provide a summary of testing completed below.
Icepack successfully creates a case, builds, runs and can be submitted to the queue using `icepack.setup -m compy ... `
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [x] No 
- [x] Please provide any additional information or relevant details below:
This PR is needed as part of the transition of E3SM to use Icepack. Note that all CICE Consortium forcing data has been copied to Compy from Cori at NERSC, and resides at /compyfs/inputdata/cice-consortium and will be maintained by @proteanplanet with rsync from NERSC.  _I should have rebased the forked master before starting, but in this instance it was up to date and doesn't matter._

### CODE REVIEW

File: configuration/scripts/icepack.batch.csh [hunk 1]
```diff
@@ -112,6 +112,22 @@ cat >> ${jobfile} << EOFB
...
+#SBATCH --ntasks ${ncores}
+#SBATCH --time ${ICE_RUNLENGTH}
+#SBATCH --cpus-per-task ${nthrds2}
+###SBATCH -e filename
+###SBATCH -o filename
+###SBATCH --mail-type FAIL
+###SBATCH --mail-user username@domain.com
+EOFB
+
+else if (${ICE_MACHINE} =~ badger*) then
```
Comment (apcraig) [inline configuration/scripts/icepack.batch.csh:130]: This is a redundant extra line and should be removed.  Will break badger (but not other machines).

File: configuration/scripts/machines/env.compy_intel [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+setenv OMP_PLACES threads
+limit coredumpsize unlimited
+limit stacksize unlimited
+
+endif
+
+setenv ICE_MACHINE_MACHNAME compy
+setenv ICE_MACHINE_MACHINFO "Intel Xeon Skylake with 192 GB of DDR4 DRAM"
+setenv ICE_MACHINE_ENVNAME intel
+setenv ICE_MACHINE_ENVINFO "ifort 19.0.3"
```
Comment (apcraig) [inline configuration/scripts/machines/env.compy_intel:31]: Propose this be
setenv ICE_MACHINE_ENVINFO "intel/19.0.3 gcc/8.1.0 intelmpi/2019u3 netcdf/4.6.3"

Comment (proteanplanet) [reply configuration/scripts/machines/env.compy_intel:31]: I have changed this to your suggested entry.

File: configuration/scripts/icepack.batch.csh (hunk 1 above)
Comment (proteanplanet) [reply configuration/scripts/icepack.batch.csh:130]: Nice catch, thank you.  It is now fixed.

**Review by apcraig — APPROVED** [review]
Thanks.  I have access to compy and will test independently when I have a chance.  We can merge before that though.

**Comment by apcraig** [comment]
I ran a full test suite on compy, everything looks good.

https://github.com/CICE-Consortium/Test-Results/wiki/icepack_by_hash_forks#0f0169e632ebd947390e95df2f698defabaec2fc

**Pull request merged**